zqd/api/logwriter.go

```diff
@@ -0,0 +1,115 @@
+package api
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"io"
+	"mime/multipart"
+	"sync"
+	"sync/atomic"
+
+	"github.com/brimsec/zq/pkg/iosrc"
+	"github.com/brimsec/zq/zio/ndjsonio"
+)
+
+type LogWriter struct {
+	URIs       []iosrc.URI
+	BytesTotal int64
+
+	bytesRead int64
+	form      *multipart.Writer
+	pr        *io.PipeReader
+	pw        *io.PipeWriter
+	start     sync.Once
+	json      *ndjsonio.TypeConfig
+}
+
+func NewLogWriter(files ...string) (*LogWriter, error) {
+	pr, pw := io.Pipe()
+	form := multipart.NewWriter(pw)
+	lw := &LogWriter{form: form, pr: pr, pw: pw}
+	for _, f := range files {
+		u, err := iosrc.ParseURI(f)
+		if err != nil {
+			return nil, err
+		}
+		info, err := iosrc.Stat(context.Background(), u)
+		if err != nil {
+			return nil, err
+		}
+		lw.BytesTotal += info.Size()
+		lw.URIs = append(lw.URIs, u)
+	}
+	return lw, nil
+}
+
+func (l *LogWriter) SetJSONTyping(config *ndjsonio.TypeConfig) {
```

Comment:
Nit: I think we should drop the "typing" terminology entirely and call this SetJSONTypeConfig or SetJSONConfig.